ROADMAP.md

```diff
@@ -0,0 +1,167 @@
+  expect the research effort itself to be built on top of KerasNLP. This enables
+  us to focus on usability and API standardization, and produce objects that
+  have a longer lifespan than the average research project.
+
+- **KerasNLP is not a repository of blackbox end-to-end solutions.**
+  KerasNLP is focused on modular and reusable building blocks. In the process
+  of developing these building blocks, we will by necessity implement
+  end-to-end workflows, but they're intended purely for demonstration and
+  grounding purposes, they're not our main deliverable.
+
+- **KerasNLP is not a repository of low-level string ops, like tf.text.**
+  KerasNLP is fundamentally an extension of the Keras API: it hosts Keras
+  objects, like layers, metrics, or callbacks. Low-level C++ ops should go
+  directly to [Tensorflow Text](https://www.tensorflow.org/text) or
+  core Tensorflow.
+
+- **KerasNLP is not a Transformer only library.**
+  Transformer based models are a key offering for KerasNLP, and they should be
+  easy to train and use within the library. However, we need to support other
+  types of models, such as n-gram or word2vec approaches that might run more
+  easily on limited hardware. We will always want the most practical tool for
+  the task, regardless of architecture.
+
+## Focus areas for 2022
+
+At this point in our development cycle, we are primarily interested in providing
+building blocks for a short list of "greatest hits" NLP models (such as BERT,
+GPT-2, word2vec). Given a popular model architecture (e.g. a
+sequence-to-sequence transformer like T5) and a end-to-end task (e.g.
+summarization), we should have a clear code example in mind and a list of
+components to use.
+
+Below, we describe our areas of focus for the year in more detail.
+
+### Easy-to-use and feature-complete tokenization
+
+KerasNLP should be the "go-to" tokenization solution for Keras model training
+and deployment by the end of 2022.
+
+The major tasks within this effort:
+
+- Work with Tensorflow Text to continue to support a growing range of
+  tokenization options and popular vocabulary formats. For example, we would
+  like to add support for byte-level BPE tokenization (the RoBERTa and GPT
+  tokenizer) within the Tensorflow graph.
+- Pre-trained sub-word tokenizers for any language. Training a tokenizer can
+  add a lot of friction to a project, particularly when working working in a
+  language where examples are less readily available. We would like to support
+  a pre-trained tokenization offering that allows a user to choose a tokenizer,
+  language, and vocabulary size and then download an off the shelf vocabulary.
+- A standardized way to training tokenizer vocabularies. As another way to
+  reduce the friction of training a tokenizer, we should offer a standardized
+  experience for training new vocabularies.
+
+### Scalable examples of popular model architectures using KerasNLP
```

Comment:
"scalable" -> should we mention DTensor support somewhere? This will be important for model parallelism.